### PR TITLE
Fix/when api reach the rate limit retry after 60 seconds.

### PR DESCRIPTION
**Pull request by oussamaessaji**
- This PR purpose is to refactor the agent in which the API rate limit occurs to retry after 1 minute as the virustotal FAQ encourages that https://support.virustotal.com/hc/en-us/articles/115002119845-What-is-the-difference-between-the-public-API-and-the-private-API- 
- also this PR fix the unit tests that are slow by mocking `sleep`

### CODE REVIEW

File: agent/virus_total_agent.py [hunk 1]
```diff
@@ -33,7 +37,7 @@ def __init__(
             agent_settings: Settings of running instance of the agent.
         """
         super().__init__(agent_definition, agent_settings)
-        self.api_key = self.args.get("api_key")
+        self.api_key = self.args.get("api_key", "")
```
Comment (3asm) [inline agent/virus_total_agent.py:40]: Why?

File: agent/virus_total_agent.py [hunk 2]
```diff
@@ -55,23 +59,46 @@ def process(self, message: msg.Message) -> None:
...
-                self._process_response(response, target)
+                self._scan_url(target)
+
+    @tenacity.retry(
+        stop=tenacity.stop_after_attempt(NUMBER_RETRIES),
+        wait=tenacity.wait_fixed(WAIT_TIME),
+        retry=tenacity.retry_if_exception_type(virustotal.VirusTotalApiError),
+        retry_error_callback=lambda retry_state: None,
+    )
+    def _scan_file(self, file_content: bytes, target: str | None) -> None:
```
Comment (3asm) [inline agent/virus_total_agent.py:74]: Why the retry?

**Review by 3asm — CHANGES_REQUESTED** [review]

File: agent/virus_total_agent.py (hunk 2 above)
Comment (oussamaessaji) [reply agent/virus_total_agent.py:74]: Retry after a minute if we reach the rate limit instead of not scanning the target!

Comment (oussamaessaji) [reply agent/virus_total_agent.py:74]: why? The rate limit is 4 per minute we need just wait another minute.

**Review by amine3 — APPROVED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**